datastructures/linkedlist/cyclicallylinkedlist/cyclicallylinkedlist_test.go

```diff
@@ -0,0 +1,108 @@
+package cyclicallylinkedlist
+
+import (
+	"reflect"
+	"testing"
+)
+
+type testCase struct {
+	param        int
+	wantToReturn int
+	listCount    int
+}
+
+func fillList(list *ClList, n int) {
+	for i := 1; i <= n; i++ {
+		list.Add(i)
+	}
+}
+
+// 100% coverage.
+func TestLinkedList(t *testing.T) {
+	list := NewList()
+	list.Add(1)
+	list.Add(2)
+	list.Add(3)
+
+	t.Run("Test Add()", func(t *testing.T) {
+		want := []interface{}{1, 2, 3}
+		var got []interface{}
+		var start *ClNode
+		start = list.currentItem
+
+		for i := 0; i < list.size; i++ {
+			got = append(got, start.val)
+			start = start.next
+		}
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("got: %v, want: %v", got, want)
+		}
+	})
+
+	t.Run("Test Walk()", func(t *testing.T) {
+		want := 1
+		got := list.Walk()
+
+		if got.val != want {
+			t.Errorf("got: %v, want: nil", got)
+		}
+	})
+
+	t.Run("Test Rotate()", func(t *testing.T) {
+		testCases := []testCase{
```

Comment:
make this global in the package - save some extra processing time in the Test function (I think)

Reply:
These cases are only for the rotate method